scvi/external/solo/_model.py

```diff
@@ -12,18 +12,18 @@
 from scvi import _CONSTANTS
 from scvi.compose import auto_move_data
 from scvi.data import get_from_registry, setup_anndata
-from scvi.dataloaders import AnnDataLoader
+from scvi.dataloaders import DataSplitter
 from scvi.lightning import ClassifierTrainingPlan
 from scvi.model import SCVI
-from scvi.model.base import BaseModelClass
+from scvi.model.base import BaseModelClass, TrainRunner, UnsupervisedTrainingMixin
 from scvi.modules import Classifier
 
 logger = logging.getLogger(__name__)
 
 LABELS_KEY = "_solo_doub_sim"
 
 
-class SOLO(BaseModelClass):
+class SOLO(UnsupervisedTrainingMixin, BaseModelClass):
```

Comment:
why does it need the mixin?

Reply:
it should only need it if you call super below, which seems like a fine option to me